typing/ikinds/ikind.mli

```diff
@@ -0,0 +1,92 @@
+(*-------------------------------------------------------------------------*)
+(* Types shared by ikind algorithms                                        *)
+(*-------------------------------------------------------------------------*)
+
+module RigidName : sig
+  (* Rigid names are the variables that may occur in ikind formulas. *)
```

Comment:
Two notes about doc comments here:
1. Starting comments with `(**` rather than `(*` makes them be a doc comment. The biggest advantage to this is than Merlin will then show the documentation when appropriate.
2. Doc comments attached to modules go on the line before `module ...`, not after